tests/tck-build-logic/src/main/groovy/org.graalvm.internal.tck-harness.gradle

```diff
@@ -102,6 +103,51 @@ def matrixDefault = [
 
 final String METADATA_GROUP = "Metadata"
 
+List<String> getNewerVersionsFor(String library, String startingVersion) {
+    def baseUrl = "https://repo1.maven.org/maven2"
+    String group = library.split(":")[0].replace(".", "/")
+    String artifact = library.split(":")[1]
+    def data = new URL(baseUrl + "/" + group + "/" + artifact + "/" + "maven-metadata.xml").getText()
+
+    return tck.getNewerVersionsFromLibraryIndex(data, startingVersion, library)
+}
+
+// gradle getExistingLibrariesWithNewVersions
+Provider<Task> getExistingLibrariesWithNewVersions = tasks.register("getExistingLibrariesWithNewVersions", DefaultTask) { task ->
+    task.setDescription("Returns list of all libraries coordinates")
+    task.setGroup(METADATA_GROUP)
+    task.doFirst {
+        Set<String> libraries = []
+        matchingCoordinates.forEach {
+            libraries.add(it.substring(0, it.lastIndexOf(":")))
+        }
+
+        List<String> newerVersions = new ArrayList<>()
+        libraries.forEach {
+            String libraryName = it
+            if (!libraryName.contains("samples") && !libraryName.contains("example")) {
+                List<String> versions = getNewerVersionsFor(libraryName, tck.getLatestLibraryVersion(libraryName))
+                versions.forEach {
+                    newerVersions.add(libraryName.concat(":").concat(it))
+                }
+            }
+        }
+
+        // save github resources as workflows has limitation to only accept up to 255 parallel jobs
+        if (newerVersions.size() > 200) {
```

Comment:
Please put 200 as a parameter. Don't link the body of this function with current limits on github.